platforms/ide/problems/src/main/java/org/gradle/problems/internal/services/ProblemsBuildSessionServices.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.problems.internal.services;
+
+import org.gradle.api.problems.internal.ExceptionProblemRegistry;
+import org.gradle.internal.service.Provides;
+import org.gradle.internal.service.ServiceRegistrationProvider;
+import org.gradle.internal.service.scopes.Scope;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+@ServiceScope(Scope.BuildSession.class)
+public class ProblemsBuildSessionServices implements ServiceRegistrationProvider {
+
+    @Provides
+    ExceptionProblemRegistry createBuildSessionProblemContainer() {
```

Comment:
❌ I don't think it makes sense for this registry to be build session scoped. The problems are only valid during one build tree. Making in build session scoped without any cleanup means that in a continuous build session you'd see the failures from the previous build invocation as well. Is that intentional?